messenger/custom-transport.rst

```diff
@@ -0,0 +1,138 @@
+How to Create Your own Messenger Transport
```

Comment:
This document was extracted to its own doc, but is unchanged and still untested.